lib/thor/actions/inject_into_file.rb

```diff
@@ -56,7 +56,11 @@ def invoke!
           replacement + '\0'
         end
 
-        replace!(/#{flag}/, content, config[:force])
+        if File.exist?(destination)
+          replace!(/#{flag}/, content, config[:force])
+        else
+          raise MalformattedArgumentError, "The file #{ destination } does not appear to exist"
```

Comment:
This exception is from when the argument can't be parsed. This should raise `Thor::Error`.